packages/@lwc/engine-dom/src/renderer.ts

```diff
@@ -222,6 +222,18 @@ export const renderer: Renderer<Node, Element> = {
         return element.getElementsByClassName(names);
     },
 
+    getNativeShadowRoot(node: Node): Node | null {
+        let ancestorElm: Node | undefined = node.getRootNode();
```

Comment:
It would be great if I could just call the actual native `getRootNode()` here, but I don't think I can because we would need some way for `@lwc/synthetic-shadow` to pass the unpatched function to `@lwc/engine-dom`, and that doesn't seem possible.